e2e/affinities/affinities.go

```diff
@@ -0,0 +1,228 @@
+package affinities
+
+import (
+	"testing"
+	"time"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/nomad/e2e/framework"
+	"github.com/hashicorp/nomad/helper"
+	"github.com/hashicorp/nomad/helper/uuid"
+	"github.com/hashicorp/nomad/jobspec"
+	"github.com/stretchr/testify/require"
+
+	. "github.com/onsi/gomega"
+)
+
+type BasicAffinityTest struct {
+	framework.TC
+	jobIds []string
+}
+
+func init() {
+	framework.AddSuites(&framework.TestSuite{
+		Component:   "Affinity",
+		CanRunLocal: true,
+		Cases: []framework.TestCase{
+			new(BasicAffinityTest),
+		},
+	})
+}
+
+func (tc *BasicAffinityTest) TestSingleAffinities(f *framework.F) {
+	nomadClient := tc.Nomad()
+
+	// Parse job
+	job, err := jobspec.ParseFile("affinities/input/aff1.nomad")
+	require := require.New(f.T())
```

Comment:
I have a small stylistic preference for this to be defined on the top of the func